Extensions created with 999-SNAPSHOT cli depend on non-existing plugin

### Describe the bug

I use Quarkus CLI to generate quarkus extension. When I use cli from 999-SNAPSHOT, the pom file contains dependency io/quarkus/quarkus-config-doc-maven-plugin of version 3.12.3, which doesn't exist.

### Expected behavior

Generated apps and extensions should not contain non-existing dependencies

### Actual behavior

```
2024-08-20T09:15:32.0278431Z [WARNING] The POM for io.quarkus:quarkus-config-doc-maven-plugin:jar:3.13.2 is missing, no dependency information available
2024-08-20T09:15:32.0279918Z [ERROR] [ERROR] Some problems were encountered while processing the POMs:
2024-08-20T09:15:32.0284197Z [ERROR] Unresolveable build extension: Plugin io.quarkus:quarkus-config-doc-maven-plugin:3.13.2 or one of its dependencies could not be resolved: io.quarkus:quarkus-config-doc-maven-plugin:jar:3.13.2 was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced @ 
2024-08-20T09:15:32.0287744Z  @ 
```

### How to Reproduce?

1. Use the latest cli:
```
          cat <<EOF > ./quarkus-dev-cli
          #!/bin/bash
          java -jar ~/.m2/repository/io/quarkus/quarkus-cli/999-SNAPSHOT/quarkus-cli-999-SNAPSHOT-runner.jar "\$@"
          EOF
          chmod +x ./quarkus-dev-cli
```
2. Create an extension using cli:
```
./quarkus-dev-cli create extension edge && cd quarkus-edge
```
3. Try to build the extension:
`mvn clean install -Dinsecure.repositories=WARN`

If `quarkus` cli of version 3.13.2 (eg from sdkman) is used in p.2, the build works

### Output of `uname -a` or `ver`

6.10.4-200.fc40.x86_64

### Output of `java -version`

Java version: 21.0.1, vendor: Eclipse Adoptium

### Quarkus version or git rev

b7d6639ec4941ec9473bd37081f51a47d2806962

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)

### Additional information
This is a difference between generated poms (`edge` is 999-SNAPSHOT, `stable` is 3.12.3):

```
$ diff quarkus-edge/pom.xml quarkus-stable/pom.xml 
10,11c10,11
<     <groupId>io.quarkiverse.edge</groupId>
<     <artifactId>quarkus-edge-parent</artifactId>
---
>     <groupId>io.quarkiverse.stable</groupId>
>     <artifactId>quarkus-stable-parent</artifactId>
14c14
<     <name>Quarkus Edge - Parent</name>
---
>     <name>Quarkus Stable - Parent</name>
23,25c23,25
<         <connection>scm:git:git@github.com:quarkiverse/quarkus-edge.git</connection>
<         <developerConnection>scm:git:git@github.com:quarkiverse/quarkus-edge.git</developerConnection>
<         <url>https://github.com/quarkiverse/quarkus-edge</url>
---
>         <connection>scm:git:git@github.com:quarkiverse/quarkus-stable.git</connection>
>         <developerConnection>scm:git:git@github.com:quarkiverse/quarkus-stable.git</developerConnection>
>         <url>https://github.com/quarkiverse/quarkus-stable</url>
62,66d61
<                 </plugin>
<                 <plugin>
<                     <groupId>io.quarkus</groupId>
<                     <artifactId>quarkus-config-doc-maven-plugin</artifactId>
<                     <version>${quarkus.version}</version>

```

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/docs/pom.tpl.qute.xml
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/docs/pom.tpl.qute.xml
@@ -55,6 +55,7 @@
                     </execution>
                 </executions>
             </plugin>
+            {#if quarkus.version.or(quarkus.bom.version).compareVersionTo("3.14") >= 0}
             <plugin>
                 <groupId>io.quarkus</groupId>
                 <artifactId>quarkus-config-doc-maven-plugin</artifactId>
@@ -63,6 +64,7 @@
                     <targetDirectory>$\{project.basedir}/modules/ROOT/pages/includes/</targetDirectory>
                 </configuration>
             </plugin>
+            {/if}
             <plugin>
                 <artifactId>maven-resources-plugin</artifactId>
                 <executions>
@@ -75,6 +77,13 @@
                         <configuration>
                             <outputDirectory>$\{project.basedir}/modules/ROOT/pages/includes/</outputDirectory>
                             <resources>
+                                {#if quarkus.version.or(quarkus.bom.version).compareVersionTo("3.14") < 0}
+                                <resource>
+                                    <directory>$\{project.basedir}/../target/asciidoc/generated/config/</directory>
+                                    <include>{namespace.id}{extension.id}.adoc</include>
+                                    <filtering>false</filtering>
+                                </resource>
+                                {/if}
                                 <resource>
                                     <directory>$\{project.basedir}/templates/includes</directory>
                                     <include>attributes.adoc</include>

--- a/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/pom.tpl.qute.xml
+++ b/independent-projects/tools/base-codestarts/src/main/resources/codestarts/quarkus-extension/code/quarkiverse/java/pom.tpl.qute.xml
@@ -8,6 +8,7 @@
         <developerConnection>scm:git:git@github.com:quarkiverse/quarkus-{extension.id}.git</developerConnection>
         <url>https://github.com/quarkiverse/quarkus-{extension.id}</url>
     </scm>
+    {#if quarkus.version.or(quarkus.bom.version).compareVersionTo("3.14") >= 0}
     <build>
         <pluginManagement>
             <plugins>
@@ -19,6 +20,7 @@
             </plugins>
         </pluginManagement>
     </build>
+    {/if}
     <profiles>
         <profile>
             <id>it</id>

--- a/integration-tests/maven/src/test/java/io/quarkus/maven/it/CreateExtensionMojoIT.java
+++ b/integration-tests/maven/src/test/java/io/quarkus/maven/it/CreateExtensionMojoIT.java
@@ -93,7 +93,7 @@ public class CreateExtensionMojoIT extends QuarkusPlatformAwareMojoTestBase {
         Properties properties = new Properties();
         properties.put("groupId", "io.quarkiverse.my-quarkiverse-ext");
         properties.put("extensionId", "my-quarkiverse-ext");
-        properties.put("quarkusVersion", "2.10.5.Final");
+        properties.put("quarkusVersion", "3.14.0");
         properties.put("extensionName", "My Quarkiverse extension");
         properties.put("extensionDescription", "My Quarkiverse extension description");
         properties.put("withCodestart", "true");
@@ -136,7 +136,7 @@ public class CreateExtensionMojoIT extends QuarkusPlatformAwareMojoTestBase {
         properties.put("groupId", "io.standalone");
         properties.put("extensionId", "my-own-ext");
         properties.put("namespaceId", "my-org-");
-        properties.put("quarkusVersion", "2.10.5.Final");
+        properties.put("quarkusVersion", "3.14.0");
         InvocationResult result = setup(properties);
 
         assertThat(result.getExitCode()).isZero();

